src/Microsoft.ML.LightGbm/LightGbmRankingTrainer.cs

```diff
@@ -222,7 +222,8 @@ private protected override void CheckDataValid(IChannel ch, RoleMappedData data)
                     $"Label column '{labelCol.Name}' is of type '{labelType.RawType}', but must be Key or Single.");
             }
             // Check group types.
-            ch.CheckParam(data.Schema.Group.HasValue, nameof(data), "Need a group column.");
+            if(!data.Schema.Group.HasValue)
+                throw ch.ExceptValue(nameof(data.Schema.Group), "Group column is missing.") ;
```

Comment:
```
if (!condition)
    throw ...
```
is the same as doing a check on `condition`.

Reply:
Hi Yael! Checkparam throws an ArgumentOutOfRangeException, I switched to checking manually, so I can throw a different type of exception. There's also a checkvalue function, but it only works for checking classes.